### PR TITLE
IBP-3147 Integrate OpenCPU with Graphical Filtering Tool

### PR DESCRIPTION
**Pull request by abatac**

### CODE REVIEW

File: src/main/java/org/ibp/api/java/impl/middleware/rpackage/RPackageServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+	private List<RCallDTO> mapToDTO(final List<RCall> rCalls) {
+		final List<RCallDTO> rCallDTOS = new ArrayList<>();
+		for (final RCall rCall : rCalls) {
+			rCallDTOS.add(this.map(rCall));
+		}
+		return rCallDTOS;
+	}
+
+	private RCallDTO map(final RCall rCall) {
```
Comment (darla-leafnode) [inline src/main/java/org/ibp/api/java/impl/middleware/rpackage/RPackageServiceImpl.java]: With the Middleware change to return DTOs, you can use ModelMapper here to transform to this another DTO

File: src/main/java/org/ibp/api/rest/rcall/RCallResource.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+@RequestMapping("/rpackage")
+public class RCallResource {
+
+	@Autowired
+	private RPackageService rPackageService;
+
+	@ApiOperation(value = "Get all R Calls", notes = "Get all R Calls")
+	@RequestMapping(value = "/rcalls", method = RequestMethod.GET)
+	@ResponseBody
+	public ResponseEntity<List<RCallDTO>> getAllRCalls() {
```
Comment (darla-leafnode) [inline src/main/java/org/ibp/api/rest/rcall/RCallResource.java]: Seems to be not used, please remove if so

File: src/main/java/org/ibp/api/rest/rcall/RCallResource.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		return new ResponseEntity<>(this.rPackageService.getAllRCalls(), HttpStatus.OK);
+	}
+
+	@ApiOperation(value = "Get R Calls by R Package", notes = "Get R Calls by R Package")
+	@RequestMapping(value = "/rcalls/{packageId}", method = RequestMethod.GET)
+	@ResponseBody
+	public ResponseEntity<List<RCallDTO>> getRCallsByPackageId(@PathVariable final Integer packageId) {
```
Comment (darla-leafnode) [inline src/main/java/org/ibp/api/rest/rcall/RCallResource.java]: For REST standards adherence, change URL to: /r-packages/{packageId}/r-calls

File: src/main/java/org/ibp/api/rest/rcall/RCallResource.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/rpackage")
+public class RCallResource {
```
Comment (darla-leafnode) [inline src/main/java/org/ibp/api/rest/rcall/RCallResource.java]: Should be renamed to RPackageResource?

**Comment by abatac** [comment]
Hi @darla-leafnode this is done. Please check. I've also done sanity testing. No need to move to branch testing again.

**Pull request merged**